presto-parquet/src/main/java/io/prestosql/parquet/reader/PrimitiveColumnReader.java

```diff
@@ -78,32 +92,80 @@ protected boolean isValueNull()
         return ParquetTypeUtils.isValueNull(columnDescriptor.isRequired(), definitionLevel, columnDescriptor.getMaxDefinitionLevel());
     }
 
-    public static PrimitiveColumnReader createReader(RichColumnDescriptor descriptor)
+    public static PrimitiveColumnReader createReader(RichColumnDescriptor descriptor, Type prestoType)
     {
-        switch (descriptor.getPrimitiveType().getPrimitiveTypeName()) {
-            case BOOLEAN:
-                return new BooleanColumnReader(descriptor);
-            case INT32:
-                return createDecimalColumnReader(descriptor).orElse(new IntColumnReader(descriptor));
-            case INT64:
-                if (descriptor.getPrimitiveType().getOriginalType() == OriginalType.TIMESTAMP_MICROS) {
-                    return new TimestampMicrosColumnReader(descriptor);
+        requireNonNull(descriptor, "descriptor is null");
+        requireNonNull(prestoType, "prestoType is null");
+
+        Optional<DecimalType> parquetDecimalType = createDecimalType(descriptor);
+        Optional<PrimitiveColumnReader> decimalColumnReader = createDecimalColumnReader(descriptor);
+        verify(parquetDecimalType.isPresent() == decimalColumnReader.isPresent(), "parquetDecimalType and decimalColumnReader should be present at the same time");
+
+        if (prestoType == BooleanType.BOOLEAN && descriptor.getPrimitiveType().getPrimitiveTypeName() == PrimitiveTypeName.BOOLEAN) {
+            return new BooleanColumnReader(descriptor);
+        }
+
+        if (prestoType == TinyintType.TINYINT
+                || prestoType == SmallintType.SMALLINT
+                || prestoType == IntegerType.INTEGER
+                || prestoType == BigintType.BIGINT) {
+            // TODO verify no truncation occurs
+            if (parquetDecimalType.isPresent() && parquetDecimalType.get().getScale() == 0) {
+                // TODO use IntColumnReader or LongColumnReader when applicable
+                return decimalColumnReader.orElseThrow(() -> new VerifyException("decimalColumnReader not present when parquetDecimalType is present"));
```

Comment:
i agree. i wrote it so because otherwise intellij reports `'Optional.get()' without 'isPresent()' check` warning.
it's false positive, but i prefer not to have even false positive warnings so that it's easier to notice the correct ones